contracts/contracts/utils/FeePayout.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: BSL 1.1
+pragma solidity 0.8.20;
+
+library FeePayout {
+
+    struct Tracker {
+        /// @dev Address that accumulates fees
+        address recipient;
+        /// @dev Accumulated fees since last payout
+        uint256 accumulatedAmount;
+        /// @dev Block number when the last fee payout was made
+        uint256 lastPayoutBlock;
+        /// @dev Min number of blocks between payouts
+        uint256 payoutPeriodBlocks;
+    }
+
+    /// @dev Event emitted when fees are transferred to the recipient.
+    event FeeTransfer(uint256 amount, address recipient);
```

Comment:
I think it will be goot to indexing recipient, so we can filter events by specific recipient addresses when search through logs, which can be very useful for tracking and auditing purposes.